### PR TITLE
Provide PHP 8.1 support 

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

- Drop php 7.3 support and add 8.1.0 per PR https://github.com/laminas/laminas-servicemanager/pull/103 by @weierophinney 
- Add `.laminas-ci.json`
- remove bump changelog
- commit composer.lock

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
         "forum": "https://discourse.laminas.dev/"
     },
     "require": {
-        "php": "^7.3 || ~8.0.0",
+        "php": "~7.4 || ~8.0.0 || ~8.1.0",
```
Comment (Ocramius) [inline composer.json:18]: We should probably have a `composer.lock`

Comment (samsonasik) [reply composer.json:18]: should we run composer update prefer lowest for it?

Comment (Ocramius) [reply composer.json:18]: Need to first try it out - I think lowest deps may have incompatibilities with 8.1

Comment (samsonasik) [reply composer.json:18]: Yes, it seems, I commited composer.lock with prefer-stable

**Comment by samsonasik** [comment]
It seems php 8.1, with latest phpstan, it still error:

```bash
PHP Fatal error:  During inheritance of ArrayAccess: Uncaught Return type of Hoa\Protocol\Node\Node::offsetExists($name) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

phar:///github/workspace/vendor/phpstan/phpstan/phpstan.phar/vendor/hoa/protocol/Node/Node.php:134
```

**Comment by samsonasik** [comment]
PHP 8.1 error possibly related with this https://github.com/phpstan/phpstan/issues/5558

**Comment by Ocramius** [comment]
Also related: https://github.com/Roave/BetterReflection/issues/701

**Comment by samsonasik** [comment]
rebased and re-generated composer.lock from php 7.4

**Comment by samsonasik** [comment]
@Ocramius All green 🎉

**Review by Ocramius — APPROVED** [review]
LGTM 👍

**Pull request merged**